src/NuGet.Core/NuGet.Protocol/Providers/RegistrationResourceV3Provider.cs

```diff
@@ -27,7 +27,11 @@ public RegistrationResourceV3Provider()
             {
                 //This will come back as null if there are no matching RegistrationsBaseUrl types
                 var baseUrl = serviceIndex.GetServiceEntryUri(ServiceTypes.RegistrationsBaseUrl);
-
+                // Check for a not HTTPS source
+                if (baseUrl.Scheme == Uri.UriSchemeHttp && baseUrl.Scheme != Uri.UriSchemeHttps && !source.PackageSource.AllowInsecureConnections)
+                {
+                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Error_HttpServiceIndexUsage, source.PackageSource.SourceUri, baseUrl));
+                }
```

Comment:
Should the code that validates the endpoint / url be promoted to the ResourceProvider so we reduce code duplication?